packages/integration-tests/src/package_manager_sanity_checks.test.ts

```diff
@@ -81,77 +81,145 @@ void describe('getting started happy path', async () => {
     if (packageManager === 'pnpm' && process.platform === 'win32') {
       return;
     }
-    if (packageManager === 'yarn-classic') {
-      await execa('yarn', ['add', 'create-amplify'], { cwd: tempDir });
-      await execaCommand('./node_modules/.bin/create-amplify --yes --debug', {
-        cwd: tempDir,
-        env: { npm_config_user_agent: 'yarn/1.22.21' },
-      });
-    } else {
-      await runPackageManager(
-        packageManager,
-        ['create', amplifyAtTag, '--yes'],
-        tempDir
-      ).run();
-    }
 
-    const pathPrefix = path.join(tempDir, 'amplify');
-
-    const files = await glob(path.join(pathPrefix, '**', '*'), {
-      nodir: true,
-      windowsPathsNoEscape: true,
-      ignore: ['**/node_modules/**', '**/yarn.lock'],
-    });
-
-    const expectedAmplifyFiles = [
-      path.join('auth', 'resource.ts'),
-      'backend.ts',
-      path.join('data', 'resource.ts'),
-      'package.json',
-      'tsconfig.json',
-    ];
-
-    assert.deepStrictEqual(
-      files.sort(),
-      expectedAmplifyFiles.map((suffix) => path.join(pathPrefix, suffix))
-    );
+    const TIMEOUT_MS = 1000 * 60 * 3; // 3 minutes
+    const startTime = Date.now();
 
-    await runWithPackageManager(
-      packageManager,
-      [
-        'ampx',
-        'pipeline-deploy',
-        '--branch',
-        branchBackendIdentifier.name,
-        '--appId',
-        branchBackendIdentifier.namespace,
-      ],
-      tempDir,
-      { env: { CI: 'true' } }
-    ).run();
-
-    const clientConfigStats = await fsp.stat(
-      await getClientConfigPath(ClientConfigFileBaseName.DEFAULT, tempDir)
-    );
+    while (Date.now() - startTime < TIMEOUT_MS) {
```

Comment:
Could you please create `retry.ts` file and create generalized function for this.

Something like

```
export const runWithRetry = async <T>(
    callable: () => Promise<T>, // code under retry
    retryPredicate: (error: Error) => boolean, // predicate to decide if retry or not
    maxAttempts = 3,
    timeout,
    ): Promise<T> => {...};
```


Similarly to this https://github.com/aws-amplify/amplify-backend/blob/06bf37e746254d49c7a6a6074449be63505f1f08/packages/integration-tests/src/test-project-setup/conversation_handler_project.ts#L967-L989 

This ^ can be then replaced with new generalized function.